### PR TITLE
Used fuzzy_bool() instead of ternary sort

### LINKED ISSUE #13699: use fuzzy_bool instead of ternary_sort
There is an existing function that can take the place of `ternary_sort`:
```diff
diff --git a/sympy/sets/sets.py b/sympy/sets/sets.py
index 659b778..f050f12 100644
--- a/sympy/sets/sets.py
+++ b/sympy/sets/sets.py
@@ -8,6 +8,7 @@
 from sympy.core.expr import Expr
 from sympy.core.singleton import Singleton, S
 from sympy.core.evalf import EvalfMixin
+from sympy.core.logic import fuzzy_bool
 from sympy.core.numbers import Float
 from sympy.core.compatibility import (iterable, with_metaclass,
     ordered, range, PY3)
@@ -217,13 +218,8 @@ def _complement(self, other):
             return S.EmptySet

         elif isinstance(other, FiniteSet):
-            from sympy.utilities.iterables import sift
-
-            def ternary_sift(el):
-                contains = self.contains(el)
-                return contains if contains in [True, False] else None
-
-            sifted = sift(other, ternary_sift)
+            sifted = sift(other,
+                lambda x: fuzzy_bool(self.contains(x)))
             # ignore those that are contained in self
             return Union(FiniteSet(*(sifted[False])),
                 Complement(FiniteSet(*(sifted[None])), self, evaluate=False)
```

### PR DESCRIPTION
**Pull request by arighnac**
Closes #13703 .

Fixes #13699 .

The fuzzy_bool() function has been implemented in the ternary_sift(e1) function.

### CODE REVIEW

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -216,7 +216,17 @@ def _complement(self, other):
             return S.EmptySet
 
         elif isinstance(other, FiniteSet):
-            return FiniteSet(*[el for el in other if self.contains(el) != True])
+            from sympy.utilities.iterable import sift
```
Comment (smichr) [inline sympy/sets/sets.py]: can these go at the top of the file?

Comment (ArighnaIITG) [reply sympy/sets/sets.py]: I didn't get you... What did you mean ?

Comment (ArighnaIITG) [reply sympy/sets/sets.py]: Can you tell me why I am having a merge conflict here?

**Comment by ArighnaIITG** [comment]
@smichr , @gxyd ,
Can you please tell me why I am having merge conflicts here?

**Comment by gxyd** [comment]
Because you checked out this branch from a somewhat old master branch. See https://github.com/sympy/sympy/blob/master/sympy/sets/sets.py this file was last changed on Nov. 20, so the branch from which you checked out this branch is possibly older than Nov. 20.

General note: Keep your local master branch updated regularly.

To fix this now, read https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/

**Comment by ArighnaIITG** [comment]
@gxyd ,
I have resolved most of my merge conflicts, but the remaining conflict consists of my code change.
How can I fix this?
Please help me.

**Comment by smichr** [comment]
When you run into a merge conflict you have to (carefully) resolve them by hand. In your case, the master has not changed, nobody else is working on this, and you just have to tell git what changes you want.

First, make a backup and then let's squash the two commits into 1:
```
git checkout 13699
git branch bak
git reset head~1; git commit --amend -a
```
This will put you into vi, just type `:wq`
Now rebase on master:
```
git rebase master
```
This will complain about the conflict. If you edit sympy/sets/sets.py you will find
```
++<<<<<<< f78eb8b8c427a62872fa2eeaca0887ba83bfc8fd
 +            def ternary_sift(el):
 +                contains = self.contains(el)
 +                return contains if contains in [True, False] else None
++=======
+             def ternary_sift(e1):
+                 return fuzzy_bool(self.contains(e1))
++>>>>>>> Used fuzzy_bool()
```
You *could* just delete the top lines and the lines with >>> and <<< to keep the bottom lines, but I would recommend taking @jksuom 's suggestion to just use a lambda in the `sift` call in which case you delete the <<< line and >>> line and everything in between and just modify the sift line to read
```
sifted = sift(other, lambda x: fuzzy_bool(self.contains(x)))
```
exit the editor with `:wq` again and tell git what you modified then continue with the rebase:
```
git add sympy/sets/sets.py
git rebase --continue
```
The push the results with `-f` and delete the backup:
```
git push -f  ArighnaIITG 13699_issue
git branch -D bak
```
No need to close this issue or reopen a new one.

**Comment by ArighnaIITG** [comment]
Thank you @smichr.

The merge conflict has been removed.
Some unnecessary commits are showing up, which are not changing any files, and they are also not appearing in my git log --oneline.

**Comment by smichr** [comment]
I think the extra commits show up when you do not have an up-to-date master. Try the following

```
git branch bak
git reset head~2
git diff > dif
git checkout master
git branch -D 13699_issue
git pull
git apply dif
git checkout -b 13699_issue
git commit 'use fuzzy_bool'
git push -f ArighnaIITG 13699_issue
git branch -D bak
```
It is going to complain when you try to apply the diff, but it seems to work anyway.

**Comment by smichr** [comment]
Please add "closes #13703" (without quotes) to your opening paragraph and rewrite "Fixes Issue #13699" as "Fixes #13699" or else it won't [close the issue](https://tinyurl.com/autoclose-syntax).

**Comment by ArighnaIITG** [comment]
@smichr , 
I tried the above in the terminal, but it didn't work. The unnecessary commits are still showing up.

**Comment by smichr** [comment]
Maybe you made modifications to your master branch? If you pull master and check the sha of the last commit on master what do you get? I get `3707b69 Merge pull request #13787 from smichr/authors`.

**Comment by ArighnaIITG** [comment]
@smichr,
I do get `3707b69 Merge pull request #13787 from smichr/authors`.
Now, what do I do?

**Comment by ArighnaIITG** [comment]
screenshot from 2017-12-24 11-30-43

**Comment by smichr** [comment]
After doing the following I confirmed that only 1 commit exists when making the PR:

```
git checkout master
git pull
git branch -D 13699_issue
git pull ArighnaIITG
git checkout ArighnaIITG/13699_issue
git diff master > dif
git checkout master
git checkout -b 13699_issue
git apply dif
git commit -am 'used fuzzy_bool'
git push -f ArighnaIITG 13699_issue
```

**Comment by ArighnaIITG** [comment]
@smichr , I have done literally everything, but those commits still remain.
I believe if I can delete this branch, and create a new one, the issue can be solved.

So, I am closing this PR, and making another new one.

**Pull request closed without merging**